SRC/main.f90

```diff
@@ -494,25 +500,41 @@ subroutine trace_orbit(anorb, ipart)
       if (integmode <= 0) then
         call orbit_timestep_axis(z, dtaumin, dtaumin, relerr, ierr)
       else
+        if (swcoll) then  ! TODO: move this inside modules
+          anorb%si%pabs = z(4)
+          anorb%f%vpar = z(4)*z(5)*sqrt2
+          anorb%f%mu = z(4)**2*(1.d0-z(5)**2)/anorb%f%Bmod
+          anorb%si%z(4) = anorb%f%vpar*anorb%f%hph + anorb%f%Aph/anorb%f%ro0
+          call get_val(anorb%f, anorb%si%z(4)) ! for pth
+          anorb%si%pthold = anorb%f%pth
+        endif
         call orbit_timestep_sympl(anorb%si, anorb%f, ierr)
         z(1:3) = anorb%si%z(1:3)
-        z(4) = 1d0
-        z(5) = anorb%f%vpar/dsqrt(2d0)
+        z(4) = dsqrt(anorb%f%mu*anorb%f%Bmod+0.5d0*anorb%f%vpar**2)
+        z(5) = anorb%f%vpar/(z(4)*sqrt2)
```

Comment:
These two lines could actually change some behavior. They have been introduced to allow collisions.